src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2222,8 +2224,25 @@ private bool getReadonlyStaticFieldValue(CORINFO_FIELD_STRUCT_* fieldHandle, byt
             FieldDesc field = HandleToObject(fieldHandle);
             Debug.Assert(field.IsStatic);
 
+
             if (!field.IsThreadStatic && field.IsInitOnly && field.OwningType is MetadataType owningType)
             {
+                if (field.HasRva)
+                {
+                    if (field is EcmaField ecmaField)
+                    {
+                        ReadOnlySpan<byte> rvaData = ecmaField.GetFieldRvaData();
+                        if (rvaData.Length >= bufferSize && valueOffset >= 0 && valueOffset <= rvaData.Length - bufferSize)
+                        {
+                            rvaData.Slice(valueOffset, bufferSize).CopyTo(new Span<byte>(buffer, bufferSize));
+                            return true;
+                        }
+                    }
+                    return false;
+                }
+
+                Debug.Assert(valueOffset == 0); // is only used for RVA at the moment
```

Comment:
"is only used for RVA at the moment" - I do not understand what this is trying to say.